server/src/session.ts

```diff
@@ -820,24 +820,34 @@ export class Session {
     }
   }
 
-  private onFoldingRanges(params: lsp.FoldingRangeParams) {
-    if (!params.textDocument.uri?.endsWith('ts')) {
-      return null;
-    }
-
+  private onFoldingRanges(params: lsp.FoldingRangeParams): lsp.FoldingRange[]|null {
     const lsInfo = this.getLSAndScriptInfo(params.textDocument);
     if (lsInfo === null) {
       return null;
     }
-    const {scriptInfo} = lsInfo;
+    const {scriptInfo, languageService} = lsInfo;
+    const angularOutliningSpans = languageService.getOutliningSpans(scriptInfo.fileName);
```

Comment:
Surprising that we already can return "outlining spans" for control flow blocks!

Reply:
This was done in https://github.com/angular/angular/pull/52062

Reply:
Ah OK, I just missed it!